release/sns_notification.sh

```diff
@@ -0,0 +1,98 @@
+#     but you can override this if you choose. The topic must already exist, and you must have permission to list topics
+#     and publish to the intended topic. See SNS doc: https://docs.aws.amazon.com/sns/latest/dg/sns-configuring.html
+#   + A directory under NOTIFICATION_BODY_FILE_PATH that has a name that indicates what release branch you're
+#     publishing the notification about (e.g. eks-distro/tree/main/release/1-19). You must provided the directory name
+#     when running the script.
+#   + A file named 'RELEASE' in the above-mentioned release branch directory. This file must contain only the release
+#     number (e.g. "2" for 1.19.2).
+#   + A file named release-announcement.txt also in the above-mentioned release branch directory. This file must contain
+#     the intended body of the SNS message.
+# - Run ./sns_notification.sh <release branch directory name> <OPTIONAL:SNS topic name>,
+#   e.g ./sns_notification.sh 1-19 my-sns-topic
+# - When prompted to confirm that you wish to publish, enter the appropriate response to do so. All ofter response exit.
+# - If the response received from SNS did not indicate there was an error, you will be provided the SNS MessageID.
+
+RELEASE_BRANCH="${1?First required argument is release branch, e.g. 1-19}"
+SNS_TOPIC_NAME="${2:-"eks-distro-updates"}"
+
+# Get release directory
+RELEASE_BRANCH_DIRECTORY="$(git rev-parse --show-toplevel)/release/${RELEASE_BRANCH}"
+
+# Get release notification body
+NOTIFICATION_BODY_FILE_PATH="${RELEASE_BRANCH_DIRECTORY}/release-announcement.txt"
+if ! [ -s "$NOTIFICATION_BODY_FILE_PATH" ]; then
+  echo "Non-empty text for notification message body expected at ${NOTIFICATION_BODY_FILE_PATH}"
+  exit 1
+fi
+NOTIFICATION_BODY=$(<"${NOTIFICATION_BODY_FILE_PATH}")
+
+# Get release notification subject
+RELEASE_NUMBER_PATH="${RELEASE_BRANCH_DIRECTORY}/RELEASE"
+if ! [ -s "$RELEASE_NUMBER_PATH" ]; then
+  echo "Release number expected at ${RELEASE_NUMBER_PATH}"
+  exit 1
+fi
+NOTIFICATION_SUBJECT="EKS-D v${RELEASE_BRANCH/-/.}.$(<"${RELEASE_NUMBER_PATH}") Release" # e.g. EKS-D v1.19.0 Release
+
+## Get SNS topic ARN
+SNS_TOPIC_ARN=$(aws sns list-topics --query "Topics[?ends_with(TopicArn,':$SNS_TOPIC_NAME')].TopicArn | [0]" --output text)
+if [ "$SNS_TOPIC_ARN" == "None" ]; then
+  echo "Failed to send SNS notification because unable to find SNS topic'${SNS_TOPIC_NAME}'"
+  exit 1
+elif [ -z "$SNS_TOPIC_ARN" ]; then
+  echo "Failed to send SNS notification due to error when trying to get the SNS topic ARN for topic '${SNS_TOPIC_NAME}'"
+  exit 1
+fi
+
+# Confirm release notification details
+read -r -p "
+A notification with the below details will be immediately published to ALL eligible subscribers. It cannot be unsent.
+
+    - Topic ARN: $SNS_TOPIC_ARN
+    - Subject: $NOTIFICATION_SUBJECT
+    - Body: $NOTIFICATION_BODY
+
+Are you sure you want to send this notification? Enter 'publish' to confirm: " confirmation_response
```

Comment:
Requiring "publish" instead of "y" was intended to be cumbersome. Publishing to an SNS topic cannot be undone, and sending incorrect information or spamming the wrong topic results in a bad experience for the subscribers. The script creates all the values used in the notification, and this confirmation step is the only chance the user has to view and check if the ARN and notification content are correct. Cumbersome-ing up the process signals to users that they shouldn't just reflexively or absentmindedly enter "y".

We do not intent to include this script in the automation process. If the user accidentally enters "y" instead of "publish", re-running the scripts takes like a second. As long as the user doesn't confirm publishing, there is no result or impact in running the script over and over again.

Reply:
Never mind. As described in other comments, we're going in a different direction